consensus/src/round_manager.rs

```diff
@@ -1351,29 +1383,44 @@ impl RoundManager {
 
     async fn new_qc_from_order_vote_msg(
         &mut self,
-        order_vote_msg: &OrderVoteMsg,
+        verified_qc: Arc<QuorumCert>,
+        preferred_peer: Author,
     ) -> anyhow::Result<()> {
+        // If the block doesn't exist, we could ideally do sync up based on the qc.
+        // But this could trigger fetching a lot of past blocks in case the node is lagging behind.
+        // So, we just log a warning here to avoid a long sequence of block fetchs.
+        // One of the subsequence syncinfo messages will trigger the block fetch or state sync if required.
+        if self
```

Comment:
can we use the same need_fetch_for_quorum_cert for this?
```
  match self.block_store.need_fetch_for_quorum_cert(..) {
    NeedFetchResult::QCAlreadyExist  => ...,
    NeedFetchResult::QCBlockExist => ...,
  }
```

Reply:
Done.